src/function/uvc.c

```diff
@@ -138,6 +139,131 @@ struct {
 		.export = usbg_set_config_node_int,		        \
 	}
 
+int h2b(unsigned char c)
+{
+	/* check if char is a number */
+	if ((c >= '0') && (c <= '9'))
+		return c - '0';
+
+	/* check if char is upper case */
+	if ((c >= 'A') && (c <= 'Z'))
+		return c - 'A' + 10;
+
+	/* check if char is lower case */
+	if ((c >= 'a') && (c <= 'z'))
+		return c - 'a' + 10;
+
+	return -EINVAL;
+}
+
+int usbg_write_guid(const char *path, const char *name,
+		      const char *file, const char *buf)
+{
+	int ret;
+	char guid[16];
+	char *tmp = guid;
+
+	if (strlen(buf) != 32)
+		return USBG_ERROR_INVALID_PARAM;
```

Comment:
The UVC definition specifies the supported GUID formats, for example, as "32595559-0000-0010-8000-00AA00389B71" for YUY2. I'd expect that the function would expect exactly that string. That would allow to directly use the GUID from the definition instead of doing some manual conversion.

Reply:
I will change the string parsing then and take the four dashes into account.